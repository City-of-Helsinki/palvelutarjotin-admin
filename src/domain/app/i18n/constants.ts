export enum VALIDATION_MESSAGE_KEYS {
  EMAIL = 'form.validation.string.email',
  NUMBER_MAX = 'form.validation.number.max',
  NUMBER_MIN = 'form.validation.number.min',
  NUMBER_REQUIRED = 'form.validation.number.required',
  STRING_MAX = 'form.validation.string.max',
  STRING_MIN = 'form.validation.string.min',
  STRING_REQUIRED = 'form.validation.string.required',
  URL = 'form.validation.string.url',
}
