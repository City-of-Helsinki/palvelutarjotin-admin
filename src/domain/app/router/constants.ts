import type { BrowserRouterProps, MemoryRouterProps } from 'react-router-dom';

export const DEFAULT_ROUTER_PROPS = {
  future: {
    v7_startTransition: true,
  },
} as const satisfies BrowserRouterProps satisfies MemoryRouterProps;
