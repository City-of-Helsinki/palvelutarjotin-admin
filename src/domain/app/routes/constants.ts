export enum ROUTES {
  CALLBACK = '/callback',
  CREATE_EVENT = '/events/create',
  CREATE_OCCURRENCE = '/events/:id/occurrences/create',
  EDIT_OCCURRENCE = '/events/:id/occurrences/:occurrenceId/edit',
  EDIT_EVENT = '/events/:id/edit',
  ENROLLMENTS = '/events/:id/occurrences/:occurrenceId/enrollments',
  EVENT_DETAILS = '/events/:id',
  HOME = '/',
  OCCURRENCES = '/events/:id/occurrences',
  SILENT_CALLBACK = '/silent-callback',
}
