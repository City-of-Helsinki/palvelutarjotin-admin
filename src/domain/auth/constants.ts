import { LoginProviderProps } from 'hds-react';

import AppConfig from '../app/AppConfig';

// todo: delete this when new login implemented
export const API_TOKEN_ACTIONS = {
  FETCH_TOKEN_ERROR: 'FETCH_TOKEN_ERROR',
  FETCH_TOKEN_SUCCESS: 'FETCH_TOKEN_SUCCESS',
  RESET_API_TOKEN_DATA: 'RESET_API_TOKEN_DATA',
  START_FETCHING_TOKEN: 'START_FETCHING_TOKEN',
  TOKEN_FETCHED: 'TOKEN_FETCHED',
};

const providerPropertiesBase: LoginProviderProps = {
  userManagerSettings: {
    authority: AppConfig.oidcAuthority,
    client_id: AppConfig.oidcClientId,
    scope: AppConfig.oidcScope,
    response_type: AppConfig.oidcReturnType,
    redirect_uri: `${AppConfig.origin}/callback`,
    automaticSilentRenew: AppConfig.oidcAutomaticSilentRenew,
    silent_redirect_uri: `${AppConfig.origin}/silent_renew.html`,
    post_logout_redirect_uri: `${AppConfig.origin}/`,
  },
  debug: false,
  sessionPollerSettings: {
    pollIntervalInMs: AppConfig.oidcSessionPollerIntervalInMs,
  },
};

/**
 * HDS LoginProvider configuration when the Kultus Admin UI
 * is integrated with Tunnistamo.
 *
 * @deprecated LoginProviderProps for Tunnistamo integration are deprecated,
 * because the Tunnistamo instance will be changed to
 * the Helsinki-Profile Keycloak instance in the near future.
 */
export const tunnistamoPoviderProperties: LoginProviderProps = {
  ...providerPropertiesBase,
  apiTokensClientSettings: {
    url: `${AppConfig.oidcAuthority}api-tokens/`,
    maxRetries: 10,
    retryInterval: 1000,
  },
};

/**
 * HDS LoginProvider configuration when the Kultus Admin UI
 * is integrated with Keycloak.
 */
export const keycloakPoviderProperties: LoginProviderProps = {
  ...providerPropertiesBase,
  apiTokensClientSettings: {
    url: `${AppConfig.oidcAuthority}protocol/openid-connect/token`,
    maxRetries: 10,
    retryInterval: 1000,
    queryProps: {
      grantType: 'urn:ietf:params:oauth:grant-type:uma-ticket',
      permission: '#access',
    },
    audiences: AppConfig.oidcAudiences,
  },
};
