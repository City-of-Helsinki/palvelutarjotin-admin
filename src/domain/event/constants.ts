export enum EVENT_LANGUAGES {
  AR = 'ar',
  EN = 'en',
  FI = 'fi',
  RU = 'ru',
  SV = 'sv',
  ZH_HANS = 'zh_hans',
}
