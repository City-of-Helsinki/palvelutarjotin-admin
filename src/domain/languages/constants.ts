import { type Language, LanguageCodeEnum } from 'react-helsinki-headless-cms';

export const CMS_FINNISH_LANGUAGE_ENTRY = {
  code: LanguageCodeEnum.Fi,
  id: 'TGFuZ3VhZ2U6Zmk=',
  locale: 'fi',
  name: 'Suomi',
  slug: 'fi',
  __typename: 'Language',
} as const satisfies Language;

export const CMS_ENGLISH_LANGUAGE_ENTRY = {
  code: LanguageCodeEnum.En,
  id: 'TGFuZ3VhZ2U6ZW4=',
  locale: 'en_US',
  name: 'English',
  slug: 'en',
  __typename: 'Language',
} as const satisfies Language;

export const CMS_SWEDISH_LANGUAGE_ENTRY = {
  code: LanguageCodeEnum.Sv,
  id: 'TGFuZ3VhZ2U6c3Y=',
  locale: 'sv_SE',
  name: 'Svenska',
  slug: 'sv',
  __typename: 'Language',
} as const satisfies Language;

export const HARDCODED_CMS_LANGUAGES = [
  { ...CMS_FINNISH_LANGUAGE_ENTRY },
  { ...CMS_ENGLISH_LANGUAGE_ENTRY },
  { ...CMS_SWEDISH_LANGUAGE_ENTRY },
] as const;

/** Data taken from production CMS languages query response. */
export const HARDCODED_CMS_LANGUAGE_QUERY_RESPONSE = {
  data: {
    languages: [...HARDCODED_CMS_LANGUAGES],
  },
} as const;
