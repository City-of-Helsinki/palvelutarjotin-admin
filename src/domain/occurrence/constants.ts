import { TimeAndLocationFormFields } from './types';

export enum OCCURRENCE_URL_PARAMS {
  ENROLMENT_UPDATED = 'enrolmentUpdated',
}

export const defaultInitialValues: TimeAndLocationFormFields = {
  location: '',
  isVirtual: false,
  enrolmentNeeded: false,
  externalEnrolment: false,
  externalEnrolmentUrl: '',
  enrolmentStart: null,
  enrolmentEndDays: '',
  neededOccurrences: '',
  autoAcceptance: true,
  locationDescription: {
    fi: '',
    en: '',
    sv: '',
  },
  hasClothingStorage: false,
  hasSnackEatingPlace: false,
  outdoorActivity: false,
  hasToiletNearby: false,
  hasAreaForGroupWork: false,
  hasIndoorPlayingArea: false,
  hasOutdoorPlayingArea: false,
};
