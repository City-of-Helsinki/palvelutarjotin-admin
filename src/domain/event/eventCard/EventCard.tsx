import { IconEye, IconUser } from 'hds-react';
import * as React from 'react';
import { useTranslation } from 'react-i18next';

import IconClock from '../../../icons/IconClock';
import { PUBLICATION_STATUS } from '../../events/constants';
import { EnrolmentType } from '../../occurrence/enrolmentInfoFormPart/EnrolmentInfoFormPart';
import { getEventPlaceholderImage } from '../utils';
import styles from './eventCard.module.scss';

interface Props {
  description?: string;
  enrolmentsCount: number;
  enrolmentType: EnrolmentType;
  id: string;
  image?: string;
  name: string;
  occurrencesCount: number;
  onClick?: (id: string) => void;
  publicationStatus?: string | null;
}

const EventCard: React.FC<Props> = ({
  description,
  enrolmentsCount,
  enrolmentType,
  id,
  image,
  name,
  occurrencesCount,
  publicationStatus,
  onClick,
}) => {
  const { t } = useTranslation();

  const handleClick = () => {
    if (onClick) {
      onClick(id);
    }
  };

  const onKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
    if (event.key === 'Enter' && onClick) {
      onClick(id);
    }
  };

  const getPublicationText = () => {
    if (publicationStatus === PUBLICATION_STATUS.PUBLIC) {
      return t('events.eventCard.statusPublished');
    }

    if (publicationStatus === PUBLICATION_STATUS.DRAFT) {
      return t('events.eventCard.statusNotPublished');
    }
  };

  const getEnrolmentText = () => {
    return {
      [EnrolmentType.Internal]: t('events.eventCard.textEnrolments', {
        count: enrolmentsCount || 0,
      }),
      [EnrolmentType.External]: t('events.eventCard.textExternalEnrolments'),
      [EnrolmentType.Unenrollable]: t('events.eventCard.textUnenrollable'),
    }[enrolmentType];
  };

  return (
    <div
      role="button"
      className={styles.eventCard}
      tabIndex={0}
      onClick={handleClick}
      onKeyDown={onKeyDown}
      aria-label={t('events.eventCard.ariaLabelOpenOccurrences', {
        eventName: name,
      })}
    >
      <div
        className={styles.imageWrapper}
        data-testid={`event-image-${id}`}
        style={{
          backgroundImage: `url(${image || getEventPlaceholderImage(id)})`,
        }}
      ></div>
      <div className={styles.contentWrapper}>
        <div className={styles.titleWrapper}>
          <div className={styles.title}>{name}</div>
          <div className={styles.description}>{description}</div>
        </div>
        <div className={styles.occurrenceInfoWrapper}>
          <div className={styles.textWithIcon}>
            <IconClock />
            {t('events.eventCard.textOccurrences', {
              count: occurrencesCount || 0,
            })}
          </div>
          <div className={styles.textWithIcon}>
            <IconUser />
            {getEnrolmentText()}
          </div>
          {/* TODO: Handle rest of the cases when there API support */}
          <div className={styles.textWithIcon}>
            <IconEye />
            {getPublicationText()}
          </div>
        </div>
      </div>
    </div>
  );
};

export default EventCard;
