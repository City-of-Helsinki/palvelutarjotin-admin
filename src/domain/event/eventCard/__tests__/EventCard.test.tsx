import { render, screen } from '@testing-library/react';
import * as React from 'react';

import { PUBLICATION_STATUS } from '../../../events/constants';
import { EnrolmentType } from '../../../occurrence/enrolmentInfoFormPart/EnrolmentInfoFormPart';
import EventCard from '../EventCard';

const defaultEventCardProps = {
  id: 'id',
  name: 'TestiTapahtuma',
  enrolmentsCount: 2,
  occurrencesCount: 8,
  publicationStatus: PUBLICATION_STATUS.PUBLIC,
  image: 'test.url',
  enrolmentType: EnrolmentType.Internal,
};

it('matches snapshot', () => {
  const { container } = render(<EventCard {...defaultEventCardProps} />);

  expect(container).toMatchSnapshot();
});

// it('is accessible', async () => {
//   const { container } = render(<EventCard {...defaultEventCardProps} />);

//   const result = await axe(container);
//   expect(result).toHaveNoViolations();
// });

it('displays correct texts and handles click', () => {
  const { rerender } = render(<EventCard {...defaultEventCardProps} />);

  expect(screen.queryByRole('button')).toBeInTheDocument();
  expect(screen.queryByText(/^8 tapahtuma-aikaa$/i)).toBeInTheDocument();
  expect(screen.queryByText(/^2 ilmoittautunutta$/i)).toBeInTheDocument();
  expect(screen.queryByText(/^julkaistu$/i)).toBeInTheDocument();

  rerender(
    <EventCard
      {...defaultEventCardProps}
      publicationStatus={PUBLICATION_STATUS.DRAFT}
    />
  );

  expect(screen.queryByText(/^julkaistu$/i)).not.toBeInTheDocument();
  expect(screen.queryByText(/^ei julkaistu$/i)).toBeInTheDocument();

  const image = screen.getByTestId(`event-image-${defaultEventCardProps.id}`);
  expect(image).toHaveStyle(
    `background-image: url(${defaultEventCardProps.image});`
  );
});

it('displays event with external enrolment correctly', () => {
  render(
    <EventCard
      {...defaultEventCardProps}
      enrolmentType={EnrolmentType.External}
    />
  );
  expect(screen.queryByText(/^2 ilmoittautunutta$/i)).not.toBeInTheDocument();
  expect(
    screen.queryByText(/ilmoittautuminen muulla sivustolla/i)
  ).toBeInTheDocument();
});

it('displays unenrollable event correctly', () => {
  render(
    <EventCard
      {...defaultEventCardProps}
      enrolmentType={EnrolmentType.Unenrollable}
    />
  );
  expect(screen.queryByText(/^2 ilmoittautunutta$/i)).not.toBeInTheDocument();
  expect(screen.queryByText(/ei ilmoittautumista/i)).toBeInTheDocument();
});
