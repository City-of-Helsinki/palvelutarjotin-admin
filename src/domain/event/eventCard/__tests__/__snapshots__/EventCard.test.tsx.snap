// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    aria-label="Siirry tapahtuman TestiTapahtuma tapahtuma-aikoihin"
    class="eventCard"
    role="button"
    tabindex="0"
  >
    <div
      class="imageWrapper"
      data-testid="event-image-id"
      style="background-image: url(test.url);"
    />
    <div
      class="contentWrapper"
    >
      <div
        class="titleWrapper"
      >
        <div
          class="title"
        >
          TestiTapahtuma
        </div>
        <div
          class="description"
        />
      </div>
      <div
        class="occurrenceInfoWrapper"
      >
        <div
          class="textWithIcon"
        >
          <svg
            class=""
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill-rule="evenodd"
              stroke-width="1"
            >
              <path
                d="M11.80416,-2.30926389e-13 C18.32352,-2.30926389e-13 23.60928,5.2848 23.60928,11.80416 C23.60928,18.32352 18.32352,23.60928 11.80416,23.60928 C5.2848,23.60928 -3.37507799e-14,18.32352 -3.37507799e-14,11.80416 C-3.37507799e-14,5.2848 5.2848,-2.30926389e-13 11.80416,-2.30926389e-13 Z M11.80416,2.5296 C6.6912,2.5296 2.53056,6.69024 2.53056,11.80416 C2.53056,16.91904 6.6912,21.07968 11.80416,21.07968 C16.91808,21.07968 21.07968,16.91904 21.07968,11.80416 C21.07968,6.69024 16.91808,2.5296 11.80416,2.5296 Z M13.068768,5.842464 L13.068768,11.280864 L16.665888,14.877024 L14.878368,16.665504 L10.539168,12.328224 L10.539168,5.842464 L13.068768,5.842464 Z"
              />
            </g>
          </svg>
          8 tapahtuma-aikaa
        </div>
        <div
          class="textWithIcon"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M16.267 13a4.733 4.733 0 014.728 4.517l.005.216V22H3v-4.267a4.733 4.733 0 014.517-4.728L7.733 13h8.534zm-8.534 2a2.733 2.733 0 00-2.728 2.567L5 17.733V20h14v-2.267c0-1.416-1.107-2.636-2.46-2.726l-.163-.005L7.733 15zM12 2a5 5 0 110 10 5 5 0 010-10zm0 2a3 3 0 100 6 3 3 0 000-6z"
                fill="currentColor"
              />
            </g>
          </svg>
          2 ilmoittautunutta
        </div>
        <div
          class="textWithIcon"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 4.5c4.05 0 7.717 2.5 11 7.5-3.283 5-6.95 7.5-11 7.5S4.283 17 1 12c3.283-5 6.95-7.5 11-7.5zm0 2c-2.951 0-5.75 1.704-8.425 5.302L3.43 12l.145.198c2.624 3.53 5.368 5.237 8.258 5.3L12 17.5c2.951 0 5.75-1.704 8.425-5.302L20.57 12l-.145-.198c-2.624-3.53-5.368-5.237-8.258-5.3L12 6.5zM12 8a4 4 0 110 8 4 4 0 010-8zm0 2a2 2 0 100 4 2 2 0 000-4z"
                fill="currentColor"
              />
            </g>
          </svg>
          Julkaistu
        </div>
      </div>
    </div>
  </div>
</div>
`;
